app/src/main/java/com/example/happymeals/recipe/RecipeStorageActivity.java

```diff
@@ -52,7 +55,65 @@ protected void onCreate(Bundle savedInstanceState) {
         ArrayAdapter<String> dataAdapter = new ArrayAdapter<String>(RecipeStorageActivity.this, android.R.layout.simple_list_item_1, getResources().getStringArray(R.array.recipe_options));
         dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
         RecipeSort.setAdapter(dataAdapter);
+        RecipeSort.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                String itemSelected = adapterView.getItemAtPosition(i).toString();
+
+                //if sort by "Amount" is selected
+                 if(itemSelected.equals("Total Time")){
+                    adapter.sort(new Comparator<Recipe>() {
+                        @Override
+                        public int compare(Recipe r1, Recipe r2) {
+                            if ((r1.getPrepTime() + r1.getCookTime()) < (r2.getPrepTime() + r2.getCookTime()))
+                                return 1;
+
+                            else if ((r1.getPrepTime() + r1.getCookTime()) > (r2.getPrepTime() + r2.getCookTime()))
+                                return -1;
+
+                            else
+                                return 0;
+
+
+                        }
+
+                    });
+                    signalChangeToAdapter();
+                    dataAdapter.notifyDataSetChanged();
+                }
+
+                if(itemSelected.equals("Serving")){
+                    adapter.sort(new Comparator<Recipe>() {
+                        @Override
+                        public int compare(Recipe r1, Recipe r2) {
+                            if (r1.getServings() < r2.getServings())
+                                return 1;
+
+                            else if (r1.getServings() < r2.getServings())
+                                return -1;
+
+                            else
+                                return 0;
+
+
+                        }
+                    });
+
+
+                }
+
+
+            }
+
```

Comment:
Can you go through all your code and fix these huge spaces between brackets, please?